### ISSUE TITLE
LS simulator workflow fails with OOM errors on 2201.1.x branch

### ISSUE THREAD

**Issue by IMS94** [ballerina-platform/ballerina-lang#36527]
**Description:**
Recently the LS simulator workflow failed frequently with OOM errors on the 2201.1.x branch. Workflow runs can be found at:
https://github.com/ballerina-platform/ballerina-lang/actions/workflows/language_server_simulator.yml

Some of the failed runs:
1. https://github.com/ballerina-platform/ballerina-lang/actions/runs/2488244655
2. https://github.com/ballerina-platform/ballerina-lang/actions/runs/2479894651
3. https://github.com/ballerina-platform/ballerina-lang/actions/runs/2468242304

**Steps to reproduce:**
Since the LS simulator adds random stuff to the ballerina documents, exact source code cannot be determined. The workflow runs contain the heapdump files generated. We can analyze them.

**Affected Versions:**
2201.1.1-SNAPSHOT

**Comment by gimantha** [comment]
I tried to reproduce this locally as we cannot find the root of the OOM from the dumps. I increased the running time of the Simulator tests to 6 hours and monitored with VisualVM. I am not seeing any memory leaks for that run. Below are the results from VisualVM.

<img width="590" alt="image" src="https://user-images.githubusercontent.com/9362271/173983927-91668877-13bd-46a7-ad47-e3f255c919d5.png">

<img width="1175" alt="image" src="https://user-images.githubusercontent.com/9362271/173983981-d66442d9-70ff-483b-b55e-7dcbace7fb86.png">

My suspect is that, this is happening only for a certain path in simulation and we need to find that specific case. This can take time
